File: src/Lending.API/Domain/Borrowing.cs
namespace Lending.API.Domain;

public class Borrowing {
	public Guid Id { get; private set; } = Guid.NewGuid();
	public Guid BookId { get; private set; }
	public string BookTitle { get; private set; } = string.Empty;
	public Guid CustomerId { get; private set; }
	public string CustomerName { get; private set; } = string.Empty;
	public DateTime BorrowedAt { get; private set; } = DateTime.UtcNow;
	public DateTime? ReturnedAt { get; private set; }

	public bool IsActive => ReturnedAt == null;

	private Borrowing() { }

	public Borrowing(Guid bookId, string bookTitle, Guid customerId, string customerName) {
		BookId = bookId;
		BookTitle = bookTitle;
		CustomerId = customerId;
		CustomerName = customerName;
	}

	public void MarkReturned() {
		if (ReturnedAt != null)
			throw new DomainException("Borrowing already returned");
		ReturnedAt = DateTime.UtcNow;
	}
}

File: frontend/src/types/borrowing.ts
export interface Borrowing {
  id: string
  bookId: string
  bookTitle: string
  customerId: string
  customerName: string
  borrowedAt: string
  returnedAt?: string
  isActive: boolean
}

export interface BorrowerInfo {
  customerId: string
  customerName: string
  borrowedAt: string
}

export interface BorrowedBookSummary {
  bookId: string
  bookTitle: string
  borrowers: BorrowerInfo[]
}

export interface BorrowBookRequest {
  bookId: string
  customerId: string
}

export interface ReturnBookRequest {
  customerId: string
}
